### PR TITLE
#2519 Fix post get all meta when one value is null

### PR DESCRIPTION
**Pull request by gustavo-roganti**
**Ticket**: #2519

## Issue
As per the ticket, this solves the issue of getting all post meta via `$post->meta()` when one meta has value set to NULL
//Expected result:
array ( 'foo' => 'bar', 'baz' => NULL, )
//Actual result:
array ( 'foo' => 'bar', 'baz' => array ( 0 => NULL, ), )

## Solution
This implements the fix proposed in the issue, changing the check 
`isset( $meta[0] )` 
to
`array_key_exists(0, $meta)` 
I added a simple failing test in the test suite, then added the fix and saw the test passing.

## Impact
None, afaik. 

## Considerations
I think this is just a small bugfix, and shouldn't really impact much.

## Testing
A simple test was added, as per committed file.

### CODE REVIEW

**Review by gchtr — DISMISSED** [review]
Hey @gustavo-roganti 

Congratulations on your first pull request! You mentioned in #2519 that you don’t know how to make one. But you figured it out, great! 🥳

And thank you so much for researching this issue and taking the time to put together a solution. You even added a correct test! 

The tests didn’t pass in your pull request because of some coding standard problems. But don’t worry, I fixed this in #2644.

To me, the solutions looks correct. I just want to get a quick second opinion from @nlemoine, before I merge in both pull requests.

**Comment by nlemoine** [comment]
Looks ok to me, congrats @gustavo-rogantifor your first PR! 😄 

@gchtr I just added some tiny changes while looking at it:
- Let's use `get_metadata` instead of `call_user_func` to grab meta values since `get_post_meta`, `get_term_meta`, etc. are just shortcuts to `get_metadata`. It's easier to read and safer because it won't fail if `$object_type` is wrong (e.g. not one of `post`, `term`, `user`, `comment`) since `$object_type` is just a public property 🙈
- I added tome tests with `add_post_meta` because the same meta key can have multiple rows

**Review by nlemoine — APPROVED** [review]

**Review by gchtr — APPROVED** [review]

**Pull request merged**
